### ISSUE TITLE
Feature request: ISO/DVD to mkv

### ISSUE THREAD

**Issue by johannesdedoper** [jeanslack/Videomass#71]
Hello @jeanslack ,

Would it possible to make it possible to insert a ISO/DVD and make a mkv file without converting the streams (lossless) including all streams and subtitles? 

Cheers,
J.

**Comment by jeanslack** [comment]
@johannesdedoper 

I think I've never tried to do it (I don't remember), what problems did you encounter?

**Comment by johannesdedoper** [comment]
Well, I tried to drag and drop ISO and VIDEO_TS.IFO files. 

The IFO file (DVD) is not recognized. Of course you can drag and drop the .VOB files but these do not contain the streaminfo and subs.

The ISO file is accepted but I get errors using eg. the predefined preset within "encoding for devices" -> ABR H.264.
It starts and continuous although it shows errors:
`File 1/1 - Pass One
Source:  "/media/tr4/DRIVE I/Music Video/DVD/ZDF Kultnacht 2013 - MUSIC/ZDF_KULTNACHT_2013_MUSIC.ISO"
Destination:  "/dev/null"
[mpeg @ 0x564a515075c0] Packet corrupt (stream = 2, dts = 16384).
[mpeg @ 0x564a515075c0] stream 2 : no TS found at start of file, duration not set
[mpeg @ 0x564a515075c0] stream 3 : no TS found at start of file, duration not set
[mpeg @ 0x564a515075c0] Could not find codec parameters for stream 2 (Audio: mp2, 0 channels): unspecified frame size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[mpeg @ 0x564a515075c0] Could not find codec parameters for stream 3 (Audio: ac3, 0 channels): unspecified sample format
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[libx264 @ 0x564a515a9180] non-strictly-monotonic PTS
[mpeg @ 0x564a515075c0] New subtitle stream 0:4 at pos:3049486 and DTS:5.52727s
[mpeg @ 0x564a515075c0] New subtitle stream 0:5 at pos:3051534 and DTS:5.52727s
[mpeg2video @ 0x564a51561540] end mismatch left=95967 2085 at 0 36
[mpeg2video @ 0x564a51561540] Warning MVs not available
/media/tr4/DRIVE I/Music Video/DVD/ZDF Kultnacht 2013 - MUSIC/ZDF_KULTNACHT_2013_MUSIC.ISO: corrupt decoded frame in stream 1
[mpeg2video @ 0x564a51561540] end mismatch left=112510 83F0 at 0 36
[mpeg2video @ 0x564a51561540] Warning MVs not available
/media/tr4/DRIVE I/Music Video/DVD/ZDF Kultnacht 2013 - MUSIC/ZDF_KULTNACHT_2013_MUSIC.ISO: corrupt decoded frame in stream 1
[mpeg2video @ 0x564a51561540] end mismatch left=261458 1C5609 at 0 36
[mpeg2video @ 0x564a51561540] Warning MVs not available
/media/tr4/DRIVE I/Music Video/DVD/ZDF Kultnacht 2013 - MUSIC/ZDF_KULTNACHT_2013_MUSIC.ISO: corrupt decoded frame in stream 1`
etc.....
At the end the mkv file was empty.

Basically, the ISO format is NOT the most important, as you can unzip the ISO to a DVD (VIDEO_TS and AUDIO_TS) folder.
However, it would be nice to import the VIDEO_TS.IFO containing all DVD info concerning streams and subs.
A program that can also do this is MakeMKV: https://www.videohelp.com/software/MakeMKV

It's just a wish to have this all in one great program, but of course you have to like the idea and think it's useful too.
Cheers,
J.

**Comment by jeanslack** [comment]
.IFO files are not supported by FFmpeg. For .ISO files have you tried to mount them instead of using them directly?

**Comment by johannesdedoper** [comment]
Ok, so that's not possible. The ISO files are basically DVD files in some sort of package. So mounting will not work either.
No problem, we forget this feature.

**Comment by jeanslack** [comment]
However now I have tried to convert a DVD ISO and I get a playable file, not an empty file, so it is not true that it is impossible.

**Comment by johannesdedoper** [comment]
Ok, what preset did you use? One with one-pass or two-pass encoding?
Does the ISO contain audio, video and subs? And does it have more audio- or videostreams?

**Comment by jeanslack** [comment]
As a first test I used a simple command `-f mpeg -c copy` with a single pass using `mpeg` output format saving it as a profile on Videomass. Then I tried "encoding for devices" -> ABR H.264, but I had to set a specific mapping for the streams.

**Comment by jeanslack** [comment]
If you encounter errors on the mapping try adding and changing the parameter `-map 0:v:0` (or `-map 0:v:1`) and `-map 0:a:1` (or `-map 0:a:1` / `-map 0:a:2` etc), v = video stream, a = audio stream 

In my ISO file I don't have subtitle streams, but I have two audio streams and one video stream.

However I think for this kind of thing it is better to use HandBrake or MakeMKV as you mentioned.

**Comment by johannesdedoper** [comment]
I agree, I will use MakeMKV for this, as I like to copy all streams and subtitles to the mkv.
